File: LeetCode/Easy/0496-next-greater-element-i/0496-next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        find_big = {v: -1 for v in nums1}
        stack = []
        ret = []

        for v in nums2:
            while stack and stack[-1] < v:
                key = stack.pop()
                find_big[key] = v
            stack.append(v)
        
        for v in nums1:
            ret.append(find_big[v])
        
        return ret

            

File: LeetCode/Easy/0496-next-greater-element-i/0496-next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        int[] findBig = new int[10000];
        Stack<Integer> stack = new Stack<>();
        int[] ret = new int[nums1.length];

        for (int i = 0; i < nums2.length; ++i){
            while(!stack.empty() && stack.peek() < nums2[i]){
                int key = stack.pop();
                findBig[key] = nums2[i];
            }
            stack.push(nums2[i]);
        }
        
        for (int i = 0; i < nums1.length; ++i) ret[i] = findBig[nums1[i]] == 0 ? -1 : findBig[nums1[i]];
        
        return ret;
    }
}